pytest_operator/plugin.py

```diff
@@ -0,0 +1,108 @@
+import os
+import pytest
+import asyncio
+from pathlib import Path
+from juju.model import Model as JujuModel
+from aioify import aioify
+from subprocess import run
+
+
+def pytest_addoption(parser):
+
+    parser.addoption("--controller", action="store", help="Juju controller to use")
+
+    parser.addoption("--model", action="store", help="Juju model to use")
+
+    parser.addoption(
+        "--cloud", action="store", help="Juju cloud to use"
+    )
+
+
+class ToolsObj:
+    """Utility class for accessing juju related tools"""
+
+    def __init__(self, request):
+        self._request = request
+
+    async def _load(self):
+        request = self._request
+        self.controller_name = request.config.getoption("--controller")
+        self.model_name = request.config.getoption("--model")
+        self.cloud = request.config.getoption("--cloud")
+        self.connection = None
+        if self.model_name and self.cloud:
+            self.connection = f"{self.controller_name}:{self.model_name}"
+
+    async def run(self, cmd, *args, input=None):
+        proc = await asyncio.create_subprocess_exec(
+            cmd,
+            *args,
+            stdin=asyncio.subprocess.PIPE,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+            env=os.environ.copy(),
+        )
+
+        if hasattr(input, "encode"):
+            input = input.encode("utf8")
+
+        stdout, stderr = await proc.communicate(input=input)
+        if proc.returncode != 0:
+            raise Exception(
```

Comment:
If we use subprocess as I suggest above, this becomes a moot point. This is a very broad exception for a runtime error, the subprocess module defines runtime errors for this that would be more appropriate. Although again, I would think we're better of just using the subprocess rather than writing our own version of it.